### PR TITLE
add setup.py, rename launcher, allow extensions via namespace packages

### PR DESCRIPTION
**Pull request by looooo**

### CODE REVIEW

**Comment by IlgarLunin** [comment]
@looooo 
Wow! Cool! 

Check [this](https://github.com/IlgarLunin/PyFlow/blob/master/requirements/requirements-standalone.txt) for full list of requirements for standalone version.

And here is docs url **pyflow.readthedocs.io**
Docs are still wip, but you can add it.

And about pyflow name, looks like **PyFlow** name already taken by [this project](https://pypi.org/project/PyFlow)

We need to think about pypi name. But this is nice! Thank you!

**Comment by looooo** [comment]
> Check this for full list of requirements for standalone version.

I excluded pyside2 as qt and pyside2/pyqt should be installed with a system package-manager or conda. But it's also ok to include it and provide pyflow as a conda-package in the long run.

Also it would be nice to add additional nodes via namespace-packages in an automatic way. This is nice to easily switch between directly integrated nodes and external developed nodes. But first I have to figure out how the current "addon-system" of pyflow works ; )

**Comment by looooo** [comment]
btw.: there is another TODO:

icons are missing if pyflow is installed with:
`pip install .`

but it works in development mode:
`pip install -e .`

So either paths are not correctly configured or icons are not included in the install.

**Comment by IlgarLunin** [comment]
Now app can search packages in following locations:

1) .PyFlow/Packages
2) Paths listed in **PYFLOW_PACKAGES_PATHS** env variable (; separated string)
3) Paths listed in Additional package locations (https://pyflow.readthedocs.io/en/latest/basics.html#general) same as **PYFLOW_PACKAGES_PATHS** variable

Default package is must have. Probably we will remove stuff like math etc.

About qt binding part - is ok, but enum is must have

**Comment by looooo** [comment]
> Now app can search packages in following locations:
>  1.PyFlow/Packages

Great. I will test if namespace-packages work this way.

**Comment by IlgarLunin** [comment]
check this 
https://github.com/IlgarLunin/PyFlow/blob/master/PyFlow/__init__.py#L118
and
https://github.com/IlgarLunin/PyFlow/blob/master/PyFlow/App.py#L511
this is actually where everything is happening related to packages loading

**Comment by looooo** [comment]
> We need to think about pypi name.

what about "pyflow.core". Extensions then can be named "pyflow.numpy" for example.

> check this
https://github.com/IlgarLunin/PyFlow/blob/master/PyFlow/__init__.py#L118
this is actually where everything is happening related to packages loading

thanks, I will have a look

**Comment by looooo** [comment]
two lines are necessary to support namespace packages:
https://github.com/IlgarLunin/PyFlow/pull/49/commits/b16f39b342f2f280a6d629bb7edcf6e9ff4c021b

I tried it with this package:
https://github.com/looooo/pyflow.template

The category is listed in the pyflow gui. But the node is not yet working.

**Comment by IlgarLunin** [comment]
@looooo no need to create templates by hand. We have package wizard, which will generate package module based on user input. Try Plugins->CreatePackage

**Comment by IlgarLunin** [comment]
@looooo i think i've understood your plan just now. Correct me if i wrong. So you want to install packages using pip or conda? For example to install pyflow + pyrr user will need to call something like

```bash
pip install pyflow.core pyflow.pyrr
```

Correct?

File: setup.py [hunk 1]
```diff
@@ -20,10 +20,6 @@
     classifiers=[
         'License :: Appache-2.0'
     ],
-    install_requires=["Qt.py",
-                      "blinker",
-                      "nine",
-                      "docutils",
-                      ],
+    install_requires=["enum", "Qt.py", "blinker", "nine", "docutils"],
```
Comment (IlgarLunin) [inline setup.py]: What about python versions 3.4+ where we dont need enum ? since it is built-in

**Comment by looooo** [comment]
@IlgarLunin 
>@looooo no need to create templates by hand. We have package wizard, which will generate package module based on user input. Try Plugins->CreatePackage

Thanks I will give it a try.

> @looooo i think i've understood your plan just now. Correct me if i wrong. So you want to install packages using pip or conda?

Yes correct, thats the idea. I guess this is the way add-ons/extensions are done in python. For sure there are other ways but this way packaging extensions is easy. Also including extensions in the main/core is easy as the same directory structure is used.

File: setup.py (hunk 1 above)
Comment (looooo) [reply setup.py]: You are right. Coming soon ;)

**Comment by looooo** [comment]
@IlgarLunin What do you think about renaming the launcher script to "pyflow"? This way launching pyflow via `pyflow` is possible if installed with pip.

File: setup.py (hunk 1 above)
Comment (looooo) [reply setup.py]: - [x] done, tested with python2 and python3

**Comment by IlgarLunin** [comment]
> @IlgarLunin What do you think about renaming the launcher script to "pyflow"? This way launching pyflow via `pyflow` is possible if installed with pip.

Agree

**Comment by looooo** [comment]
ok, so I think this PR is ready. Next I will try to add a setup.py to https://github.com/wonderworks-software/PyFlowPyrr .

**Comment by IlgarLunin** [comment]
@looooo **Thank you so much**! Tested on linux ubuntu virtual machine and windows 10. Added some small tweaks, and all works like a charm! Soon i will publish it on pypi

**Pull request merged**